zerver/worker/queue_processors.py

```diff
@@ -600,24 +597,30 @@ def maybe_send_batched_emails(self) -> None:
             self.timer_event = None
 
             current_time = time.time()
-            for user_profile_id, timestamp in list(self.batch_start_by_recipient.items()):
-                if current_time - timestamp < self.BATCH_DURATION:
+
+            all_user_batches = MissedMessageEmailBatch.objects.all().select_related()
```

Comment:
Would it make sense to do this query as `.filter(next_process_time__lte=current_time)`, thus letting us remove the `continue` loop below?  

And when we make BATCH_DURATION configurable, it'll become some `Q` query for effectively 

`current_time - user_profile.batch_duration > next_process_time`

Reply:
Yes, that's definitely better. Done!

Reply:
I also renamed `next_process_time` -> `first_event_receival_time`. I think that better suggests what the field is about.